utils/mediaupload.js

```diff
@@ -119,6 +119,7 @@ export function mediaUpload( {
 					id: savedMedia.id,
 					link: savedMedia.link,
 					url: savedMedia.source_url,
+					sizes: get( savedMedia, [ 'media_details', 'sizes' ], {} ),
```

Comment:
Can we just leave the value undefined when sizes are not available? sizes: get( savedMedia, [ 'media_details', 'sizes' ] ). I think we don't need to return an empty object when sizes are not relevant.

This issue makes me question if mediaUpload should not just return the objects it receives from the server

Reply:
> This issue makes me question if mediaUpload should not just return the objects it receives from the server

Yes, I'm curious whether there was a deliberate decision to limit the object size. My preference would be to return the entire object (particularly because different media types have different attributes)

Reply:
> My preference would be to return the entire object (particularly because different media types have different attributes)

Although, if we do this now, we change the shape of the data returned to the callback.

Reply:
Another option: create `savedMedia.mediaDetails = {}`, and the only include a `sizes` attribute for images with sizes. I think this is my preferred middle ground.